### ISSUE TITLE
ARM Version for Single Board Computer

### ISSUE THREAD

**Issue by GreenLunar** [assaultcube/AC#435]
AC is perfect for old computers, laptops and SBCs.
An ARM port able to be played on SBCs is going to be great and also might increase popularity.

_-- You know, currently dhewm3 (Doom3 Source Port) is kind of too much for SBC to handle..._

**Comment by GreenLunar** [comment]
@patlefort hi

Assuming you're the person I'm referring to...
According to [comment of 2021-12-19 16:58](https://aur.archlinux.org/pkgbase/assaultcube/#comment-842107), AC should be installable on ArchLinuxARM.  Did you try it there yet?

> I added armv7h in arch list and changed binaries to be named native_client/server. I'm keeping the _srcdir because it's my standard way of packaging and usually source archives are supposed to have a sub folder. That one is just badly packaged in its archive. Previous versions had a sub folder.

**Comment by GreenLunar** [comment]
Wasn't AC 1.2 (server) compatible with ARM?
If it's easy to do, then please make supports for server so that AC would have even more servers and later port the client to ARM too.

**Comment by patlefort** [comment]
I added arm7h based on a previous comment but another comment said it doesn't work. I have no way to test myself. I'll add it back if it's officially supported.

**Comment by Calinou** [comment]
> Wasn't AC 1.2 (server) compatible with ARM?

There was no official AssaultCube 1.2 port to ARM, but Linux distribution packagers may have figured out how to build and run a server on ARM.

Note that depending on the target hardware, running a client on ARM may require an OpenGL 1.x to OpenGL ES translation layer. Most ARM SBCs do not have native OpenGL support, but some of them have a built-in translation layer for OpenGL 1.x.

**Comment by GreenLunar** [comment]
> In Raspberry Pi 4 works flawless

1. What Raspberry Pi 4 OS do you use?
2. Please pust a video, just to really see it.

**Comment by symbios24** [comment]
I have 1gb and 4gb it works in both
I dont have video for the moment

OS:Raspbian buster lite,i played it a lot

![IMG_20210917_214258](https://user-images.githubusercontent.com/10866601/182112324-ecdf98a3-8661-45fb-ab62-c797a351351b.jpg)

**Comment by GreenLunar** [comment]
@symbios24 Thank you for the image and details. Did you yet try AC on a Raspberry Pi OS (**64-bit**)?


Raspberry Pi OS Lite
Release date: April 4th 2022
System: **32-bit**
Kernel version: 5.15
Debian version: 11 (bullseye)

From https://www.raspberrypi.com/software/operating-systems/

**Comment by symbios24** [comment]
I havent try it yet on bullseye or in a 64 bit os

Im still using buster the April 4th 2022 image

Sometime before i managed to make urban terror to work if you know it,the game has way better graphics.

**Comment by GreenLunar** [comment]
On Wed, 03 Aug 2022 02:00:28 -0700
symbios24 ***@***.***> wrote:

> I havent try it yet on bullseye or in a 64 bit os
>
> Im still using buster the April 4th 2022 image
>
> Sometime before i managed to make urban terror to work if you know
> it,the game has way better graphics.
>

Am I correct to assume, you run urban terror on a Raspberry Pi too?

**Comment by symbios24** [comment]
Yes its quake 3 engine,and smoking guns,and even age of empires 2 but that one its a compination of wine/box86

**Comment by GreenLunar** [comment]
Sauerbraten was also ported to ARM and is available on phones running postmarketOS https://pkgs.alpinelinux.org/packages?name=sauerbraten

**Comment by symbios24** [comment]
Most probably it will work i havent try it yet, assault cube is good and small,i havent try the newest 13 version,they should put the build instructions for Raspberry Pi in the main GitHub page its quite complicated.
